src/org/jitsi/impl/neomedia/codec/video/vp8/DePacketizer.java

```diff
@@ -493,6 +494,32 @@ public static boolean isKeyFrame(byte[] buff, int off, int len)
         private static final byte X_BIT = (byte) 0x80;
 
         /**
+         * Gets the temporal layer index (TID), if that's set.
+         *
+         * @param buf the byte buffer that holds the VP8 packet.
+         * @param off the offset in the byte buffer where the VP8 packet starts.
+         * @param len the length of the VP8 packet.
+         *
+         * @return the temporal layer index (TID), if that's set, -1 otherwise.
+         */
+        public static int getTemporalLayerIndex(byte[] buf, int off, int len)
+        {
+            if (buf == null || buf.length < off + len || len < 2)
+            {
+                return -1;
+            }
+
+            if ((buf[off] & X_BIT) == 0 || (buf[off+1] & T_BIT) == 0)
+            {
+                return -1;
+            }
+
+            int sz = getSize(buf, off);
+
```

Comment:
ah, I'm sorry, I misread your first comment. We should expect getSize to return -1. I'll address this.